src/sync/app.hpp

```diff
@@ -58,6 +57,181 @@ class App {
+                            const std::string& password,
+                            std::function<void(Optional<AppError>)> completion_block);
+
+        /**
+         * Confirms an email identity with the username/password provider.
+         *
+         * - parameters:
+         *     - token: The confirmation token that was emailed to the user.
+         *     - token_id: The confirmation token id that was emailed to the user.
+         *     - completion_block: A callback to be invoked once the call is complete.
+         */
+        void confirm_user(const std::string& token,
+                          const std::string& token_id,
+                          std::function<void(Optional<AppError>)> completion_block);
+
+        /**
+         * Re-sends a confirmation email to a user that has registered but
+         * not yet confirmed their email address.
+         *
+         * - parameters:
+         *     - email: The email address of the user to re-send a confirmation for.
+         *     - completion_block: A callback to be invoked once the call is complete.
+         */
+        void resend_confirmation_email(const std::string& email,
+                                       std::function<void(Optional<AppError>)> completion_block);
+
+        /**
+         * Sends a password reset email to the given email address.
+         *
+         * - parameters:
+         *     - email: The email address of the user to send a password reset email for.
+         *     - completion_block: A callback to be invoked once the call is complete.
+         */
+        void send_reset_password_email(const std::string& email,
+                                       std::function<void(Optional<AppError>)> completion_block);
+
+        /**
+         * Resets the password of an email identity using the
+         * password reset token emailed to a user.
+         *
+         * - parameters:
+         *     - password: The desired new password.
+         *     - token: The password reset token that was emailed to the user.
+         *     - token_id: The password reset token id that was emailed to the user.
+         *     - completion_block: A callback to be invoked once the call is complete.
+         */
+        void reset_password(const std::string& password,
+                            const std::string& token,
+                            const std::string& token_id,
+                            std::function<void(Optional<AppError>)> completion_block);
+
+        /**
+         * Resets the password of an email identity using the
+         * password reset function set up in the application.
+         *
+         * TODO: Add an overloaded version of this method that takes
+         * TODO: raw, non-serialized args.
+         *
+         * - parameters:
+         *     - email: The email address of the user.
+         *     - password: The desired new password.
+         *     - args: A pre-serialized list of arguments. Must be a JSON array.
```

Comment:
What extra args are needed here?

Reply:
This method specifically calls a remote function that the user has set up on the Stitch server. The args are passed forward to that function.

Reply:
Ok 👍 Is the name of the function fixed?

Reply:
Yes– the function ran is chosen and setup on the backend.